### ISSUE TITLE
UAL does not differentiate between some vehicle configurations

### ISSUE THREAD

**Issue by jurisch-it** [loki79uk/FS25_UniversalAutoload#177]
There is a problem with some vehicles/trailers where UAL does not differentiate between the configurations.
For example, the “data/vehicles/riedler/timberTrailer3A/timberTrailer3A” can be configured for 2x4.5m and therefore has a longer loading area. UAL does not currently differentiate between this: selectedConfigs="ALL”

**Comment by jurisch-it** [comment]
Good to know about that, So for the timberTrailer3A this should be useConfigName=“design”?
The sizes are adjusted via designConfigurations.

**Comment by loki79uk** [comment]
> Good to know about that, So for the timberTrailer3A this should be useConfigName=“design”? The sizes are adjusted via designConfigurations.

Yes, `useConfigName=“design”` is correct.  I basically have a version ready to release that I think fixes this vehicle, but it might depend what settings you already have saved.  I'm just trying to test it now.

**Comment by loki79uk** [comment]
Should be resolved for that 3A trailer at least in the latest version, but I still need to add some menu item to change the selected config for mods where injecting like this is not an option.  https://github.com/loki79uk/FS25_UniversalAutoload/releases

**Comment by alzhur** [comment]
Looks like I've found another trailer, which require `useConfigName=“design”` trick:
https://www.farming-simulator.com/mod.php?mod_id=305523&title=fs2025
To make it work with current UAL release, I've added this to UniversalAutoloadManager.addLocalConfigIfAvailable function
```
	if configFileName == "FS25_HW80WoodTrailer/hw80.xml" then
		spec.useConfigName = "design"
	end
```
and config
```
        <vehicle configFileName="FS25_HW80WoodTrailer/hw80.xml">
            <configuration selectedConfigs="ALL|1" useConfigName="design">
                <options isLogTrailer="true"/>
                <loadingArea offset="-0.000 1.353 0.137" width="1.919651" length="6.051138" height="1.671164"/>
            </configuration>
            <configuration selectedConfigs="ALL|2" useConfigName="design">
                <options/>
                <loadingArea offset="0.000 1.365 0.137" width="2.571850" length="5.819858" height="2.162774"/>
            </configuration>
        </vehicle>
```

**Comment by alzhur** [comment]
Updated mod to v0.4.5.1, main game - to 1.6.0.0 and deleted modSettings\UniversalAutoload.xml file for clear experiment.
Results:
1. Text strings for new 3 options are missing in localization files, as result window looks like this:
![Image](https://github.com/user-attachments/assets/d7fa4c9c-29ab-4404-a7b6-df587b83a211)
2. Options working good, two configurations for HW80 Wood Trailer was successfully created without manual edit of xml file.
3. For some strange reason, this new version of UAL auto-detect loading area for this log trailer with height of 0.1 or 0.0 meters with 50|50 chance, and if height=0.0, there is no way to increase it without editing xml.
4. If config for this trailer is already saved, values for new 3 options are not loaded into shop UAL menu at all.
5. If config for this trailer is already saved, it's not applied in shop window by default, only after switching 
designs in shop UI.

Points 4 and 5 working fine for vanilla timberTrailer3A, maybe something mod-specific?

**Comment by loki79uk** [comment]
Okay, thanks.  I will add the strings (I forgot those for all languages) and it looks like I need to rewrite the logic for detecting when the configName is used.

**Comment by loki79uk** [comment]
I have added the languages, and I think simplified the settings selection (hopefully without breaking anything else) https://github.com/loki79uk/FS25_UniversalAutoload/releases/tag/v0.4.5.2

**Comment by loki79uk** [comment]
Closing as I think this is complete now, still some refinements needed with managing the different configurations that are stored, but the option to select by configuration name is now implemented.
